ariesframework/src/main/java/org/hyperledger/ariesframework/agent/Agent.kt

```diff
@@ -106,6 +108,23 @@ class Agent(val context: Context, val agentConfig: AgentConfig) {
         ledgerService.delete()
     }
 
+    /**
+     * Remove a specified credential from the wallet.
+     */
+    suspend fun deleteCredential(credentialId: String) {
+        var creds = credentialRepository.getAll()
+
+        creds.forEach {
+            val record = it
+            it.credentials.forEach {
+                if (it.credentialRecordId == credentialId) {
+                    credentialRepository.delete(record)
```

Comment:
I don't think we need to delete credential records. Credential record stores exchange status not credential itself, and it's id is different from the credential's id.
Deleting the credentials below will be enough.

Reply:
> I don't think we need to delete credential records. Credential record stores exchange status not credential itself, and it's id is different from the credential's id.
> Deleting the credentials below will be enough.

I have made the change to only delete the credential.